### PR TITLE
AMDGPU: Rename intrinsics and remove f16/bf16 versions for load transpose

### PR DESCRIPTION
**Pull request by changpeng**
Rename the intrinsics to close to the instruction mnemonic names:
Use global_load_tr_b64 and global_load_tr_b128 instead of global_load_tr.

  This patch also removes f16/bf16 versions of builtins/intrinsics. To simplify the design, we should avoid enumerating all possible types in implementing builtins. We can always use bitcast.

### CODE REVIEW

**Comment by rampitec** [comment]
> global_load_re_b64

Type global_load_re_b64.

**Comment by changpeng** [comment]
> > global_load_re_b64
> 
> Type global_load_re_b64.

Changed! Thanks.

**Comment by srpande** [comment]
There is no issue in changing the names in principle. Curious, what is the rationale to use more demangled names?

**Comment by changpeng** [comment]
> There is no issue in changing the names in principle. Curious, what is the rationale to use more demangled names?

more user friendly.

**Comment by changpeng** [comment]
ping

File: clang/lib/CodeGen/CGBuiltin.cpp [hunk 1]
```diff
@@ -18533,51 +18533,35 @@ Value *CodeGenFunction::EmitAMDGPUBuiltinExpr(unsigned BuiltinID,
   }
   case AMDGPU::BI__builtin_amdgcn_global_load_tr_b64_i32:
   case AMDGPU::BI__builtin_amdgcn_global_load_tr_b64_v2i32:
-  case AMDGPU::BI__builtin_amdgcn_global_load_tr_b128_v4bf16:
-  case AMDGPU::BI__builtin_amdgcn_global_load_tr_b128_v4f16:
   case AMDGPU::BI__builtin_amdgcn_global_load_tr_b128_v4i16:
-  case AMDGPU::BI__builtin_amdgcn_global_load_tr_b128_v8bf16:
-  case AMDGPU::BI__builtin_amdgcn_global_load_tr_b128_v8f16:
   case AMDGPU::BI__builtin_amdgcn_global_load_tr_b128_v8i16: {
```
Comment (srpande) [inline clang/lib/CodeGen/CGBuiltin.cpp:18538]: unnecessary line here.

Comment (changpeng) [reply clang/lib/CodeGen/CGBuiltin.cpp:18538]: Maybe. People may have good reason to place it there, and this patch should not touch it.

**Review by srpande — APPROVED** [review]
LGTM

**Pull request merged**